http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/leonardoServiceContractModel.scala

```diff
@@ -298,3 +300,62 @@ object GetRuntimeResponse {
     diskConfig
   )
 }
+
+final case class CreateAppRequest(kubernetesRuntimeConfig: Option[KubernetesRuntimeConfig],
+                                  appType: AppType,
+                                  diskConfig: Option[DiskConfigRequest],
+                                  labels: LabelMap = Map.empty)
+
+final case class GetAppResponse(kubernetesRuntimeConfig: KubernetesRuntimeConfig,
+                                errors: List[KubernetesError],
+                                status: AppStatus, //TODO: do we need some sort of aggregate status?
+                                proxyUrls: Map[ServiceName, URL],
+                                diskName: Option[DiskName])
+
+object GetAppResponse {
+  def fromDbResult(appResult: GetAppResult): GetAppResponse = {
+    val errors = appResult.cluster.errors ++ appResult.nodepool.errors ++ appResult.app.errors
+    GetAppResponse(
+      KubernetesRuntimeConfig(
+        appResult.nodepool.numNodes,
+        appResult.nodepool.machineType,
+        appResult.nodepool.autoscalingEnabled
+      ),
+      errors,
+      if (errors.isEmpty) appResult.app.status else AppStatus.Error,
+      Map.empty, //TODO: Implement when proxy functionality exists
+      appResult.app.appResources.disk.map(_.name)
+    )
+  }
+}
+
+final case class ListAppResponse(googleProject: GoogleProject,
+                                 kubernetesRuntimeConfig: KubernetesRuntimeConfig,
+                                 errors: List[KubernetesError],
+                                 status: AppStatus, //TODO: do we need some sort of aggregate status?
+                                 proxyUrls: Map[ServiceName, URL],
+                                 appName: AppName,
+                                 diskName: Option[DiskName])
+
+object ListAppResponse {
+  def fromCluster(c: KubernetesCluster): List[ListAppResponse] =
+    c.nodepools.flatMap(n =>
+      n.apps.map { a =>
+        val errors = c.errors ++ n.errors ++ a.errors
+        ListAppResponse(
+          c.googleProject,
+          KubernetesRuntimeConfig(
+            n.numNodes,
+            n.machineType,
+            n.autoscalingEnabled
+          ),
+          errors,
+          if (errors.isEmpty) a.status else AppStatus.Error, //TODO: aggregate?
```

Comment:
so, for now, based on discussions with Adrian, I think the UI will only care about the app status, unless there is an error somewhere else. We may want to aggregate in a more intelligent way down the line though